MediaBrowser.Common/Plugins/BasePlugin.cs

```diff
@@ -56,7 +56,8 @@ public abstract class BasePlugin : IPlugin, IPluginAssembly
         /// Gets a value indicating whether the plugin can be uninstalled.
         /// </summary>
         public bool CanUninstall => !Path.GetDirectoryName(AssemblyFilePath)
-            .Equals(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), StringComparison.InvariantCulture);
+            .Equals(Path.GetDirectoryName(Assembly.GetExecutingAssembly().Location), StringComparison.InvariantCulture)
+            && !typeof(IPluginServiceRegistrator).IsAssignableFrom(this.GetType());
```

Comment:
Why?

Reply:
Because as it stands, plugins using this method of DI cannot be uninstalled, else they would leave a "hanging" pointer in the DI subsystem, as i haven't found a way to unregister them.

Reply:
Unless, of course, uninstalled means "uninstalled on next restart", in which case the additional line can be removed.

Reply:
Uninstalling plugins already always requires a restart

Reply:
removed.